core/services/chainlink/application.go

```diff
@@ -373,16 +377,24 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 				globalLogger,
 				streamRegistry,
 				pipelineRunner,
-				cfg.JobPipeline()),
-			job.Workflow: workflows.NewDelegate(
-				globalLogger,
-				registry,
-				legacyEVMChains,
+				cfg.JobPipeline(),
 			),
 		}
 		webhookJobRunner = delegates[job.Webhook].(*webhook.Delegate).WebhookJobRunner()
 	)
 
+	var peerID string
+	if externalPeerWrapper != nil {
+		peerID = externalPeerWrapper.GetPeer().ID().String()
```

Comment:
hmm does that work? Peer is initialized only when the wrapper service is started. You might need to pass the whole wrapper object and call it later.

I wonder if it makes sense to move "myID" to the Registry at some point...